geest/core/workflows/default_index_score_workflow.py

```diff
@@ -0,0 +1,216 @@
+import os
+import glob
+from qgis.core import (
+    QgsMessageLog,
+    Qgis,
+    QgsFeedback,
+    QgsFeature,
+    QgsVectorLayer,
+    QgsField,
+    QgsGeometry,
+    QgsRectangle,
+    QgsRasterLayer,
+    QgsProject,
+)
+from qgis.PyQt.QtCore import QVariant
+import processing  # QGIS processing toolbox
+from .workflow_base import WorkflowBase
+
+
+class DefaultIndexScoreWorkflow(WorkflowBase):
+    """
+    Concrete implementation of a 'Use Default Index Score' workflow.
+    """
+
+    def __init__(self, attributes: dict, feedback: QgsFeedback):
+        """
+        Initialize the TemporalAnalysisWorkflow with attributes and feedback.
+        :param attributes: Dictionary containing workflow parameters.
+        :param feedback: QgsFeedback object for progress reporting and cancellation.
+        """
+        super().__init__(attributes, feedback)
+
+    def execute(self):
+        """
+        Executes the workflow, reporting progress through the feedback object and checking for cancellation.
+        """
+
+        QgsMessageLog.logMessage(
+            "Executing Use Default Index Score", tag="Geest", level=Qgis.Info
+        )
+        QgsMessageLog.logMessage(
+            "----------------------------------", tag="Geest", level=Qgis.Info
+        )
+        for item in self.attributes.items():
+            QgsMessageLog.logMessage(
+                f"{item[0]}: {item[1]}", tag="Geest", level=Qgis.Info
+            )
+        QgsMessageLog.logMessage(
+            "----------------------------------", tag="Geest", level=Qgis.Info
+        )
+
+        # loop through self.bboxes_layer and the self.areas_layer  and create a raster mask for each feature
+        index_score = self.attributes["Default Index Score"]
+        for feature in self.bboxes_layer.getFeatures():
+            geom = feature.geometry()  # todo this shoudl come from the areas layer
+            aligned_box = geom
+            mask_name = f"bbox_{feature.id()}"
+            self.create_raster(
+                geom=geom,
+                aligned_box=aligned_box,
+                mask_name=mask_name,
+                index_score=index_score,
+            )
+        # TODO Jeff copy create_raster_vrt from study_area.py
+        # Create and add the VRT of all generated raster masks if in raster mode
+        self.create_raster_vrt(
+            output_vrt_name=os.path.join(self.workflow_directory, "combined_mask.vrt")
+        )
+
+        steps = 10
+        for i in range(steps):
```

Comment:
dont forget to incorporate the cancel logic in your own workflow and then remove this